internal/provider/resource_locally_signed_cert.go

```diff
@@ -70,7 +72,7 @@ func createLocallySignedCert(d *schema.ResourceData, meta interface{}) error {
 	if err != nil {
 		return err
 	}
-	caKey, err := parsePrivateKey(d, "ca_private_key_pem", "ca_key_algorithm")
+	caKey, _, err := parsePrivateKeyPEM([]byte(d.Get("ca_private_key_pem").(string)))
```

Comment:
Similar note about getting the algorithm here and calling `d.Set("ca_key_algorithm", caKeyAlgorithm)`

```suggestion
	caKey, caKeyAlgorithm, err := parsePrivateKeyPEM([]byte(d.Get("ca_private_key_pem").(string)))
```